src/java.base/share/native/libjli/java.c

```diff
@@ -525,9 +526,42 @@ JavaMain(void* _args)
     appClass = GetApplicationClass(env);
     NULL_CHECK_RETURN_VALUE(appClass, -1);
 
+    /*  Get launcher helper class. */
+    helperClass = GetLauncherHelperClass(env);
+
+    /*  Validate and fetch main method from application class. */
+    validateMainMethod = (*env)->GetStaticMethodID(env, helperClass,
+                                         "validateMainMethod",
+                                         "(Ljava/lang/Class;)Ljava/lang/reflect/Method;");
+    CHECK_EXCEPTION_NULL_LEAVE(validateMainMethod);
+    mainMethod = (*env)->CallStaticObjectMethod(env, helperClass, validateMainMethod, mainClass);
+    NULL_CHECK_RETURN_VALUE(mainMethod, -1);
+
+    /*
+     * validateMainMethod ensures that the main method's signature is correct,
+     * therefore further checking is not required.
+     */
+#define MAIN_WITHOUT_ARGS 1
+#define MAIN_NONSTATIC 2
+
+    /*
+     * getMainType determines whether the main method is static and whether it has a
+     * String[] argument.
+     */
+    jmethodID getMainType = (*env)->GetStaticMethodID(env, helperClass,
+                                                      "getMainType",
+                                                      "(Ljava/lang/reflect/Method;)I");
+    CHECK_EXCEPTION_NULL_LEAVE(getMainType);
+    mainType = (*env)->CallStaticIntMethod(env, helperClass, getMainType, mainMethod);
+    CHECK_EXCEPTION_LEAVE(-1);
+
     /* Build platform specific argument array */
-    mainArgs = CreateApplicationArgs(env, argv, argc);
-    CHECK_EXCEPTION_NULL_LEAVE(mainArgs);
+    if ((mainType & MAIN_WITHOUT_ARGS) == 0) {
+        mainArgs = CreateApplicationArgs(env, argv, argc);
+        CHECK_EXCEPTION_NULL_LEAVE(mainArgs);
```

Comment:
side: this is just a comment not proposing any change, I know it is a good practice to check for nulls here, but having a null here is really possible? how can we have a null here?

Reply:
Yes it's a combo thing. This was the code that was there BTW.

Reply:
yes I know it is previous code, I was just wondering how we could have a null exception here but it is true that other exceptions can occur